packages/x-components/src/__stubs__/related-tags-stubs.factory.ts

```diff
@@ -9,7 +9,7 @@ import { RelatedTag } from '@empathyco/x-types';
  *
  * @internal
  */
-export function getRelatedTagsStub(amount = 9): RelatedTag[] {
+export function getRelatedTagsStub(amount = 12): RelatedTag[] {
```

Comment:
Was the issue that there weren't enough related tags, and because of that the sliding panel wasn't able to perform a second scroll? That's what I've understood from the PR and from trying to break it myself, but I'm not sure I've got it right.

In any case I was also testing removing the `wait` just for fun, and slightly changing the `visible related tags have changed` allows us to do so. The change is just to use a `.should` inside the scroll `.then`, instead of another `then`, and  mappping the related tags there. This won't wait 100 ms, Cypress will keep re-executing the code until at least 1 related tag has changed or the 30s timeout happens.

```ts
Then('visible related tags have changed', () => {
  cy.getByDataTest('sliding-panel-scroll').then($scroll => {
    cy.getByDataTest('related-tag-item')
      .filter((_index, $element) => isElementVisible($element, $scroll.get(0)))
      .should($newRelatedTags => {
        const newRelatedTags = $newRelatedTags
          .map((_index, $relatedTag) => $relatedTag.textContent ?? '')
          .toArray();
        const areRelatedTagsDifferent = initialRelatedTags.some(relatedTag => !newRelatedTags.includes(relatedTag));
        expect(areRelatedTagsDifferent).to.be.true;
      });
  });
});
```

Reply:
I'm not sure either as with 9 RTs still works.

Reply:
For what I have seen I don't even know why it is/was working with 9 RTs. When manually tested, with only one click in the `sliding-panel-right-button` the end of the RTs is reached, so why is a second click possible?